src/ripple/rpc/impl/RPCHelpers.cpp

```diff
@@ -651,28 +681,50 @@ keypairForSignature(Json::Value const& params, Json::Value& error)
                 std::string(jss::key_type) + " is used.");
             return { };
         }
-
-        seed = getSeedFromRPC (params, error);
     }
-    else
+
+    // ripple-lib encodes seed used to generate an Ed25519 wallet in a
+    // non-standard way. While we never encode seeds that way, we try
+    // to detect such keys to avoid user confusion.
+    if(seed = RPC::parseRippleLibSeed(params))
```

Comment:
```suggestion
    if ((seed = RPC::parseRippleLibSeed(params)))
```
Add redundant params arround assignment in `if` when not declaring a new variable.